airflow/utils/log/file_task_handler.py

```diff
@@ -218,9 +225,17 @@ def close(self):
         if self.handler:
             self.handler.close()
 
-    def _render_filename(self, ti: TaskInstance, try_number: int) -> str:
+    def _render_filename(self, ti_or_ti_key: TaskInstance | TaskInstanceKey, try_number: int) -> str:
         """Returns the worker log filename."""
+        from airflow.models.taskinstance import TaskInstance, TaskInstanceKey
+
         with create_session() as session:
+            if isinstance(ti_or_ti_key, TaskInstanceKey):
+                ti = TaskInstance.get_from_key(ti_or_ti_key, session)
+                if not ti:
+                    raise ValueError("TaskInstance not found")
```

Comment:
It is better to raise an Airflow exception to inform the user that this exception is raised by Airflow and not by a 3rd party library.